src/main/java/org/openjdk/shenandoah/ShenandoahVisualizer.java

```diff
@@ -358,6 +394,8 @@ public abstract static class Render {
 
         final JFrame frame;
 
+        ArrayList<RegionPopUp> popups = new ArrayList<RegionPopUp> ();
```

Comment:
Declared type should be `List<RegionPopUp>` (always use the most generic type you can for the declaration).